### PR TITLE
Glossary View OMAS - client added

### PR DESCRIPTION
**Pull request by lcpopa**
Signed-off-by: Liviu Popa <liviu-constantin.popa@ing.com>

### CODE REVIEW

File: open-metadata-implementation/access-services/glossary-view/glossary-view-api/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/exception/GlossaryViewOmasException.java [hunk 1]
```diff
@@ -7,14 +7,14 @@
 /**
  * Stores checked OMRS exceptions info until it is sent to the client. Made to be used only by {@code OMRSClient}
  */
-public class OMRSExceptionWrapper extends OMRSCheckedExceptionBase {
+public class GlossaryViewOmasException extends OMRSCheckedExceptionBase {
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/glossary-view/glossary-view-api/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/exception/GlossaryViewOmasException.java]: OMAS Exceptions should extend the common services exceptions or OCF

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java [hunk 2]
```diff
@@ -0,0 +1,599 @@
...
+    private static final String GET_TRANSLATIONS = PATH_ROOT + "/terms/{2}/translations-terms" + PARAM_DELIMITER + PAGINATION;
+    private static final String GET_IS_A = PATH_ROOT + "/terms/{2}/is-a" + PARAM_DELIMITER + PAGINATION;
+    private static final String GET_VALID_VALUES = PATH_ROOT + "/terms/{2}/valid-values" + PARAM_DELIMITER + PAGINATION;
+    private static final String GET_USED_IN_CONTEXTS = PATH_ROOT + "/terms/{2}/used-in-contexts" + PARAM_DELIMITER + PAGINATION;
+    private static final String GET_ASSIGNED_ELEMENTS = PATH_ROOT + "/terms/{2}/assigned-elements" + PARAM_DELIMITER + PAGINATION;
+    private static final String GET_ATTRIBUTES = PATH_ROOT + "/terms/{2}/attributes" + PARAM_DELIMITER + PAGINATION;
+    private static final String GET_SUBTYPES = PATH_ROOT + "/terms/{2}/subtypes" + PARAM_DELIMITER + PAGINATION;
+    private static final String GET_TYPES = PATH_ROOT + "/terms/{2}/types" + PARAM_DELIMITER + PAGINATION;
+
+    /**
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java]: Add constructor with no userId and password

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java [hunk 3]
```diff
@@ -0,0 +1,599 @@
...
+    public GlossaryViewClient(String serverName, String serverPlatformRootURL, String userId, String password) throws
+            org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException {
+        super(serverName, serverPlatformRootURL, userId, password);
+    }
+
+    /**
+     * Extract all glossary definitions
+     *
+     * @param userId calling user
+     * @param serverName instance to call
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java]: Not all parameters are included in Javadoc

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java [hunk 4]
```diff
@@ -0,0 +1,599 @@
...
+            throws PropertyServerException, InvalidParameterException, GlossaryViewOmasException {
+
+        return getSingleEntityResponse("getGlossary", GET_GLOSSARY, serverName, userId, glossaryGUID);
+    }
+
+    /**
+     * Extract a term's home glossary
+     *
+     * @param userId calling user
+     * @param serverName instance to call
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java]: ServerName should be in constructor - this applied to other calls.

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java [hunk 5]
```diff
@@ -0,0 +1,599 @@
...
+     * @param glossaryGUID glossary GUID
+     *
+     * @return EntityDetailResponse terms
+     *
+     * @throws PropertyServerException if a problem occurs while serving the request
+     * @throws InvalidParameterException if parameter validation fails
+     * @throws GlossaryViewOmasException if a problem occurs on the omas backend
+     */
+    public GlossaryViewEntityDetailResponse getTermsOfGlossary(String serverName, String userId, String glossaryGUID,
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java]: The result of the client calls should be a standard Java interface with the beans returned and the exceptions called out.  The response object contains the exception information which should be returned as exceptions

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/OmasClient.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+                                                                                     String userId, String entityGUID, Integer from, Integer size)
+            throws PropertyServerException, InvalidParameterException, GlossaryViewOmasException{
+
+        invalidParameterHandler.validateUserId(userId, methodName);
+        invalidParameterHandler.validateGUID(entityGUID, "entityGUID", methodName);
+
+        GlossaryViewEntityDetailResponse response = callGetRESTCall(methodName, GlossaryViewEntityDetailResponse.class,
+                serverPlatformURLRoot + path, serverName, userId, entityGUID, from, size);
+
+        exceptionHandler.detectAndThrowOMRSException(methodName, response);
```
Comment (mandy-chessell) [inline open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/OmasClient.java]: Not all exceptions are from OMRS - infact there should be NO OMRS exceptions returned to an OMAS client.  The exceptions returned should be common services, OMAS or OCF exceptions.

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java (hunk 2 above)
Comment (lcpopa) [reply open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java]: Added

File: open-metadata-implementation/access-services/glossary-view/glossary-view-api/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/exception/GlossaryViewOmasException.java (hunk 1 above)
Comment (lcpopa) [reply open-metadata-implementation/access-services/glossary-view/glossary-view-api/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/exception/GlossaryViewOmasException.java]: Changed extended class

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java (hunk 3 above)
Comment (lcpopa) [reply open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java]: Revised all methods in the client and added the javadoc for the missing params

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java (hunk 4 above)
Comment (lcpopa) [reply open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java]: Removed serverName from method signatures and switched usage to field

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/OmasClient.java (hunk 6 above)
Comment (lcpopa) [reply open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/OmasClient.java]: Changed method name to reflect the change of exception type

File: open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java (hunk 5 above)
Comment (lcpopa) [reply open-metadata-implementation/access-services/glossary-view/glossary-view-client/src/main/java/org/odpi/openmetadata/accessservices/glossaryview/client/GlossaryViewClient.java]: Changed return types from wrapper to specific pojos

**Comment by lcpopa** [comment]
Corrected code based on previous observations, but I have also introduced some small modifications:
- setting the type returned to 'ExternalGlossaryLink', since it was not being returned by targeted calls
- extracted fields to classes for better readability and access to logic
- pom updates
- renamed some classes for consistency
- simplified exception handling
- java doc update
- code improvements as a result of sonar scan

**Pull request merged**